esmvalcore/preprocessor/_derive/gpp_grid.py

```diff
@@ -7,8 +7,21 @@
 class DerivedVariable(DerivedVariableBase):
     """Derivation of variable `gpp_grid`."""
 
-    # Required variables
-    required = [{'short_name': 'gpp', 'fx_files': ['sftlf']}]
+    @staticmethod
+    def required(project):
+        """Declare the variables needed for derivation."""
+        # Required variables
+        if project == 'CMIP5':
```

Comment:
It looks like the CMIP5 and CMIP6 requirements are the same, so just make it one case by doing something like
`if project in 'CMIP5', 'CMIP6':`